### PR TITLE
feat(condo): DOMA-4182 added validation for creating a duplicate cont…

### PR DESCRIPTION
**Pull request by VKislov**
…act on contact/create page
After filling in all the required fields, a request is sent to receive a contact with the specified `propertyId, unitName, unitType, name, phone, organization`, as well as a check that the user has not been deleted (`deletedAt: null`). If at least one contact comes from the server, `Contact with this phone number is already registered in this room` will be displayed in the container with errors. 
Screenshot 2022-09-15 at 15 30 27

### CODE REVIEW

File: apps/condo/domains/contact/components/ContactsEditor/useContactsEditorHook.tsx [hunk 1]
```diff
@@ -47,12 +56,60 @@ export const useContactsEditorHook = ({ organization, role, allowLandLine }: ICo
         setShouldCreateContact(isNew)
     }
 
+    const [findContactDuplicate, { loading: isLoadingContactDuplicate  }] = useLazyQuery(LOAD_CONTACTS, {
+        onCompleted: (data) => {
+            setContactDuplicate(get(data, 'result', []))
+            setIsContactDuplicateFetched(true)
+        },
+        onError: (e) => {
```
Comment (pahaz) [inline apps/condo/domains/contact/components/ContactsEditor/useContactsEditorHook.tsx:64]: It’s wrong and to broad check.

**Review by pahaz — COMMENTED** [review]
It looks like a little hacky. It’s better to process the case with error and ignore the specific error case

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 2]
```diff
@@ -110,17 +113,37 @@ export const CreateContactForm: React.FC = () => {
...
+    }
+
     return (
         <FormWithAction
-            action={action}
+            action={actionWithHandleSubmit}
             layout='horizontal'
             validateTrigger={['onBlur', 'onSubmit']}
             colon={false}
+            onChange={()=> setIsFieldsChanged(true) }
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:144]: useCallback

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 3]
```diff
@@ -257,6 +280,11 @@ export const CreateContactForm: React.FC = () => {
                                         ({ getFieldsValue, getFieldError }) => {
                                             const { phone, property, unitName, name } = getFieldsValue(['phone', 'property', 'unitName', 'name'])
                                             const propertyMismatchError = getFieldError('property').find((error)=>error.includes(AddressNotSelected))
+                                            const hasDuplicateError = Boolean(getFieldError('_NON_FIELD_ERROR_').find((error => error.includes(ContactDuplicateError))))
+                                            const hasContactDuplicate = isFieldsChanged ? false : hasDuplicateError
+                                            if (hasDuplicateError) {
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:285]: remove empty if block

File: apps/condo/domains/contact/schema/Contact.js [hunk 4]
```diff
@@ -9,14 +9,33 @@ const { UNIT_TYPE_FIELD } = require('@condo/domains/common/schema/fields')
...
+        messageForUser: 'api.contact.CONTACT_DUPLICATE_ERROR',
+    },
+    UNABLE_TO_UPDATE_CONTACT_DUPLICATE: {
+        mutation: 'signinResidentUser',
+        code: BAD_USER_INPUT,
+        type: UNABLE_TO_UPDATE_CONTACT_DUPLICATE,
+        message: 'Cannot update contact, because another contact with the same provided set of "property", "unitName", "name", "phone"',
+        messageForUser: 'api.contact.CONTACT_DUPLICATE_ERROR',
+    },
+
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/contact/schema/Contact.js:37]:
```suggestion
```

**Review by nomerdvadcatpyat — APPROVED** [review]

**Pull request merged**